src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,829 @@
+                    {
+                        if (syntaxFacts.IsInvocationExpression(invocation))
+                        {
+                            var invocations = methodCallSitesBuilder[refLocation.Document];
+                            invocations = invocations.Add((TInvocationExpressionSyntax)invocation);
+                            methodCallSitesBuilder[refLocation.Document] = invocations;
+                        }
+                    }
+                }
+            }
+            return methodCallSitesBuilder.ToImmutable();
+        }
+
+        /// <summary>
+        /// If the parameter is optional and the invocation does not specify the parameter, then
+        /// a named argument needs to be introduced.
+        /// </summary>
+        private static SeparatedSyntaxList<SyntaxNode> AddArgumentToArgumentList(
+            SeparatedSyntaxList<SyntaxNode> invocationArguments, SyntaxGenerator generator,
+            SyntaxNode newArgumentExpression, int insertionIndex, string name, bool named)
+        {
+            var argument = named
+                ? generator.Argument(name, RefKind.None,
+                    newArgumentExpression.WithAdditionalAnnotations(Simplifier.Annotation))
+                :
+                generator.Argument(newArgumentExpression.WithAdditionalAnnotations(Simplifier.Annotation));
+
+            return invocationArguments.Insert(insertionIndex, argument);
+        }
+
+        /// <summary>
+        /// Generates a method declaration containing a return expression of the highlighted expression.
+        /// Example:
+        /// public void M(int x, int y)
+        /// {
+        ///     int f = x * y; // highlight this expression
+        /// }
+        /// 
+        /// ---------------------------------------------------->
+        /// 
+        /// public int M_f(int x, int y)
+        /// {
+        ///     return x * y;
+        /// }
+        /// 
+        /// public void M(int x, int y)
+        /// {
+        ///     int f = x * y;
+        /// }
+        /// </summary>
+        private static async Task GenerateNewMethodAsync(Document document, TExpressionSyntax expression,
+            IMethodSymbol methodSymbol, SyntaxNode containingMethod, string newMethodIdentifier, SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            // Remove trailing trivia because it adds spaces to the beginning of the following statement.
+            var newStatements = SpecializedCollections.SingletonEnumerable(editor.Generator.ReturnStatement(expression.WithoutTrailingTrivia()));
+
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var typeSymbol = semanticModel.GetTypeInfo(expression, cancellationToken).ConvertedType ?? semanticModel.Compilation.ObjectType;
+            var newMethod = CodeGenerationSymbolFactory.CreateMethodSymbol(methodSymbol, name: newMethodIdentifier, returnType: typeSymbol);
```

Comment:
CreateMethodSymbol already can take in the statements you want to add to it.
Consider calling ICodeGenerationService instead.  It's more full featured and will support the users chosen code style (like expression bodied members).  Please update to that and add test for supporting the users style here.

Reply:
updated to that, I think I need to retrieve the "Options" from somewhere because just passing in Options.Default throws an error